agent/command/git.go

```diff
@@ -393,240 +434,260 @@ func (c *gitFetchProject) Execute(ctx context.Context, comm client.Communicator,
-					revision = module.Branch
-					c.logModuleRevision(logger, revision, moduleName, "defaulting to HEAD for merge")
-				} else {
-					revision = patchModule.Githash
-					if revision != "" {
-						c.logModuleRevision(logger, revision, moduleName, "specified in set-module")
-					}
-				}
-			}
-		}
-		if revision == "" {
-			revision = c.Revisions[moduleName]
-			if revision != "" {
-				c.logModuleRevision(logger, revision, moduleName, "specified as parameter to git.get_project")
-			}
-		}
-		if revision == "" {
-			revision = conf.Expansions.Get(moduleRevExpansionName(moduleName))
-			if revision != "" {
-				c.logModuleRevision(logger, revision, moduleName, "from manifest")
-			}
-		}
-		// if there is no revision, then use the revision from the module, then branch name
-		if revision == "" {
-			if module.Ref != "" {
-				revision = module.Ref
-				c.logModuleRevision(logger, revision, moduleName, "ref field in config file")
-			} else {
-				revision = module.Branch
-				c.logModuleRevision(logger, revision, moduleName, "branch field in config file")
-			}
-		}
-		var owner, repo string
-		owner, repo, err = thirdparty.ParseGitUrl(module.Repo)
-		if err != nil {
-			logger.Execution().Error(err.Error())
-		}
-		if owner == "" || repo == "" {
-			continue
-		}
-
-		var modulePatch *patch.ModulePatch
-		if p != nil {
-			// find module among the patch's Patches
-			for i := range p.Patches {
-				if p.Patches[i].ModuleName == moduleName {
-					modulePatch = &p.Patches[i]
-					break
-				}
-			}
-		}
-
-		opts := cloneOpts{
-			location: module.Repo,
-			owner:    owner,
-			repo:     repo,
-			branch:   "",
-			dir:      moduleBase,
-		}
-		// Module's location takes precedence over the project-level clone
-		// method.
-		if strings.Contains(opts.location, "git@github.com:") {
-			opts.method = distro.CloneMethodLegacySSH
-		} else {
-			opts.method = projectMethod
-			opts.token = projectToken
-		}
-		if err = opts.validate(); err != nil {
-			return errors.Wrap(err, "could not validate options for cloning")
-		}
-
-		var moduleCmds []string
-		moduleCmds, err = c.buildModuleCloneCommand(conf, opts, revision, modulePatch)
+		err = c.fetchModuleSource(ctx, conf, logger, jpm, opts.method, opts.token, p, moduleName)
 		if err != nil {
-			return err
-		}
-
-		err = jpm.CreateCommand(ctx).Add([]string{"bash", "-c", strings.Join(moduleCmds, "\n")}).
-			Directory(filepath.ToSlash(filepath.Join(conf.WorkDir, c.Directory))).
-			SetOutputSender(level.Info, logger.Task().GetSender()).SetErrorWriter(stdErr).Run(ctx)
-
-		errOutput := stdErr.String()
-		if errOutput != "" {
-			if opts.token != "" {
-				errOutput = strings.Replace(errOutput, opts.token, "[redacted oauth token]", -1)
-			}
-			logger.Execution().Info(errOutput)
-		}
-		if err != nil {
-			return errors.Wrap(err, "problem with git command")
+			logger.Execution().Error(err)
 		}
 	}
 
+	// Apply additional patches for commit queue batch execution.
```

Comment:
Is there a reason we handle modules in between fetching / applying the additional patches? (If we aren't sure I respect not re-factoring this until later if it might be breaking but it does seem like it makes more sense to have these happen together)

Reply:
I wondered about this too and was avoiding changing it purely because this is a refactor.